pages-domain/src/main/java/nl/vpro/domain/page/RelationDefinitionServiceProvider.java

```diff
@@ -0,0 +1,31 @@
+package nl.vpro.domain.page;
+
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * @author Michiel Meeuwissen
+ * @since 5.15.4
+ */
+@Slf4j
+public class RelationDefinitionServiceProvider {
+
+    private RelationDefinitionServiceProvider() {
+
+    }
+
+    private static RelationDefinitionService instance;
+
+    public static void setInstance(RelationDefinitionService instance) {
+        RelationDefinitionService previous = RelationDefinitionServiceProvider.instance;
+        RelationDefinitionServiceProvider.instance = instance;
+        if (previous == null || instance.equals(previous)) {
+            log.info("Set relation definition service to {}", instance);
+        } else {
+            log.warn("Replaced relation definition service {} -> {}", previous, instance);
```

Comment:
This is another solution to tackle multiplicity. I would always throw an error if it is being overwritten to guard against unnoticed and inconsistent behavior. Should it be allowed to overwrite an instance?

Reply:
Ok, let's throw, I don't have a case for overwriting right away.